cpp/test/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 # warning C4310: cast truncates constant value
 cxx_add_compile_options(MSVC /wd4310)
 
+# Core unit tests don't build under VS2013 without this.
+cxx_add_compile_options(MSVC /EHa)
```

Comment:
This is needed for Boost 1.53 compatibility (and possibly 1.54-1.58 as well). It's scoped to just the unit tests, so we're OK for now.